### ISSUE TITLE
[FIX-026] Preserve FP64 accumulation in RMSNorm

### ISSUE THREAD

**Issue by thekaveh** [thekaveh/NNx#261]
**Category:** Numerics and quantization · **Priority:** P3 · **Size:** S

**Evidence status.** Reproduced numerical defect in the second review; source is unchanged at the reviewed revision. The dramatic failure requires large finite FP64 inputs, so this is lower priority than the existing reduced-precision failures.

**Context and description.** [RMSNorm.forward](https://github.com/thekaveh/NNx/blob/0c707eb5b80d2173843e60102ba326e4f997749d/src/nnx/nn/net/transformer_layers.py#L39) unconditionally converts inputs to FP32 before squaring and reducing. Promoting half-precision inputs to FP32 is useful, but demoting FP64 needlessly loses precision and representable range. The executed second-review probe passed an FP64 tensor filled with `1e30` to a double-precision, unit-weight RMSNorm and received all zeros. FP64 represents both the input and its square; the expected normalized output is approximately all ones. The FP32 square overflows and its inverse root becomes zero. Casting this result back to FP64 cannot recover it.

**Scope.** Choose a reduction dtype that is at least FP32 and preserves FP64 input accumulation. Keep the established output and learned-weight dtype semantics. Validate both forward and backward against direct references.

**Out of scope.** A new normalization family, arbitrary extreme-magnitude overflow resistance beyond the chosen accumulation dtype, accelerator benchmarking, or a library-wide precision-policy redesign ([FEAT-028](https://github.com/thekaveh/NNx/issues/248)).

**Acceptance criteria.**

- [ ] A unit-weight FP64 layer applied to a finite `1e30` constant tensor returns finite values approximately equal to one, using a documented double-precision tolerance.
- [ ] Ordinary FP64 inputs produce outputs and input/weight gradients matching an FP64 reference without intermediate FP32 rounding.
- [ ] FP16/BF16 retain at least FP32 reduction; supported forward/backward fixtures remain finite and their output dtype contract is unchanged.
- [ ] Existing FP32 shape, RMS, learned-weight and Transformer integration tests continue to pass, with no parameter or checkpoint-schema changes.

**Reproducer and expected result (existing API).**

```python
import torch
from nnx.nn.net.transformer_layers import RMSNorm

layer = RMSNorm(4).double()
x = torch.full((1, 4), 1e30, dtype=torch.float64)
y = layer(x)
torch.testing.assert_close(y, torch.ones_like(y), rtol=1e-12, atol=1e-12)
# Currently fails: y contains zeros. It should be approximately ones.
```

**Scope and integration notes.** None. Independent of [FIX-004](https://github.com/thekaveh/NNx/issues/203)'s attention dtype mismatch and [FIX-015](https://github.com/thekaveh/NNx/issues/225)'s DoRA denominator underflow. [FEAT-028](https://github.com/thekaveh/NNx/issues/248) should consume this corrected primitive rather than make its repair contingent on the new policy.

## Delivery order and dependencies

- **Execution order:** 62 / 67 (priority-aware topological order).
- **Dependency wave:** 1. Items in the same wave may proceed independently when their prerequisites are satisfied.
- **Blocked by:** None within this backlog.
- **Related / optional integration:** [FIX-004](https://github.com/thekaveh/NNx/issues/203), [FIX-015](https://github.com/thekaveh/NNx/issues/225), [FEAT-028](https://github.com/thekaveh/NNx/issues/248). These links do not block this ticket.

**Dependency rationale.** Independent repair of current behavior. None. Independent of FIX-004's attention dtype mismatch and FIX-015's DoRA denominator underflow. FEAT-028 should consume this corrected primitive rather than make its repair contingent on the new policy.

Reviewed against [`0c707eb`](https://github.com/thekaveh/NNx/commit/0c707eb5b80d2173843e60102ba326e4f997749d) on 2026-09-20. Acceptance criteria are prospective; this issue does not claim an implementation or new test pass.
